archiver/integrity.py

```diff
@@ -41,6 +51,47 @@ def check_integrity(source_path, deep_flag=False, threads=None, work_dir=None):
     return check_result
 
 
+def check_archive_part_integrity(source_name: Path) -> bool:
+
+    check_result = True
+    for s in [HASH_SUFFIX, TAR_HASH_SUFFIX, LISTING_SUFFIX]:
+        path = source_name.parent / Path(source_name.name + s)
+        if not path.is_file():
+            logging.warning(f"Expected file {path.as_posix()} does not exists.")
+            check_result = False
+    path_c = source_name.parent / Path(source_name.name + COMPRESSED_ARCHIVE_SUFFIX)
+    path_e = source_name.parent / Path(source_name.name + ENCRYPTED_ARCHIVE_SUFFIX)
+    if not (path_c.is_file() or path_e.is_file()):
+        logging.warning(f"Neither of the expected files {path_c.as_posix()} and {path_e.as_posix()} exist.")
+        check_result = False
+
+    path_c = source_name.parent / Path(source_name.name + COMPRESSED_ARCHIVE_HASH_SUFFIX)
```

Comment:
There is some code duplication with the previous block, could be extracted in a small private helper method

Reply:
Did not put it into a helper, but removed most of the redundancy with a loop.